src/librustc_mir/interpret/validity.rs

```diff
@@ -782,7 +782,7 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValueVisitor<'mir, 'tcx, M>
                     Err(err) => {
                         // For some errors we might be able to provide extra information
                         match err.kind {
-                            err_ub!(InvalidUndefBytes(Some(ptr))) => {
+                            err_ub!(InvalidUndefBytes { ptr: Some(ptr), .. }) => {
                                 // Some byte was uninitialized, determine which
```

Comment:
We could also generate a range by using the size to detect whether multiple elements are unintialized. Probably not worth the effort though, especially since it's only moderately helpful if there are initialized pieces in between.